Create tutorial for Model Registry (#1307)
VertaAI/modeldb: client/verta/docs/tutorials/packaging/version_from_run.rst

> Creating Model Version From Experiment Run

Reviewer: What's the motivation? Even a single line can help the user understand if this will be useful or not. Right now they need to read through and try to figure out if it's useful for their case or not. I'd recommend following the template structure to ensure this if covered.

Author: Basically, according to my understanding, the motivation is that during the experimentation process, data scientists will log stuff to experiment run.

After that, the ones good enough to production will be moved to the model registry. Verta provides a convenient way to convert an experiment run into a model version for this purpose.

I'm going to summarize this as "providing a direct link between the experimentation process and production", and also add a "context" section to explain more about the scenario.

Creating model versions from experiment runs
============================================

> An alternative to creating model versions `from artifacts <version_from_artifacts.html>`_ is to convert an existing experiment run into a new version for the registered model. This conversion provides a direct link between the experimentation process and production.

Reviewer 2: ```suggestion
An alternative to creating model versions `from artifacts <version_from_artifacts.html>`_ is to convert an existing experiment run into a new version of a registered model. This conversion provides a direct link between the experimentation process and production.
```

In this tutorial, we will explore how this can be done, using the client and the CLI.

Scenario
--------

> During experimentation, data scientists can use Verta to log the model and deployment-specific information (requirements, custom modules, model API, etc.):

Reviewer 2: ```suggestion
During experimentation, data scientists would use Verta to log a trained model along with associated artifacts and metadata:
```

.. code-block:: python

    experiment_run.log_model(model=model, custom_modules=custom_modules, model_api=model_api)
    experiment_run.log_requirements(["torch==1.0.0"])

> The models that are good enough for production now needs to be moved to the model registry, as versions of a registered model.

Reviewer 2: ```suggestion
The models that are good enough for production can now be moved to the model registry, as a version of a registered model.
```

> The new versions should automatically inherit requirements, artifacts, model, etc. from the experiment runs they were created from.

Reviewer 2: Be specific, here. "Should" isn't very reassuring in a tutorial!
```suggestion
A model version created from an experiment run will automatically inherit its model, artifacts, Python requirements, and training histogram.
```

Using the client
----------------

> To create a new model version from an existing experiment run, we can use :meth:`RegisteredModel.create_version_from_run() <verta._registry.model.RegisteredModel.create_version_from_run>` method:

Reviewer 2: ```suggestion
To create a new model version from an existing experiment run, we can use :meth:`RegisteredModel.create_version_from_run() <verta._registry.model.RegisteredModel.create_version_from_run>`:
```

.. code-block:: python

>     model_version = registered_model.create_version_from_run(
>         run_id=experiment_run.id
>         name="from-experiment-run"
>     )

Reviewer 2: ```suggestion
    model_version = registered_model.create_version_from_run(
        run_id=experiment_run.id,
        name="from-experiment-run",
    )
```

Using the CLI
-------------

With Verta's CLI, we can create a new model version from an existing experiment run with the ``from-run`` option:

.. code-block:: sh

>     verta registry create registeredmodelversion <model name> <version name> --from-run <experiment run id>

Reviewer 2: ```suggestion
    verta registry create registeredmodelversion "<model name>" "<version name>" --from-run "<experiment run id>"
```
